backend/sftp/sftp.go

```diff
@@ -984,10 +984,10 @@ func NewFsWithConnection(ctx context.Context, f *Fs, name string, root string, m
 				outString := string(bytes.TrimSpace(stdout.Bytes()))
 				if strings.HasPrefix(outString, "Microsoft.PowerShell") { // If PowerShell: "Microsoft.PowerShell%ComSpec%"
 					f.shellType = "powershell"
-				} else if !strings.HasSuffix(outString, "%ComSpec%") { // If Command Prompt: "${ShellId}C:\WINDOWS\system32\cmd.exe"
+				} else if outString != "" && !strings.HasSuffix(outString, "%ComSpec%") { // If Command Prompt: "${ShellId}C:\WINDOWS\system32\cmd.exe"
 					f.shellType = "cmd"
-				} else { // If Unix: "%ComSpec%"
-					f.shellType = "unix"
+				} else { // If POSIX-based shell: "%ComSpec%", if fish shell: ""
```

Comment:
Do we actually need the `${ShellId}` at all? We don't seem to be using it.

Reply:
It is used for this rule:
```go
if strings.HasPrefix(outString, "Microsoft.PowerShell") { // If PowerShell: "Microsoft.PowerShell%ComSpec%"
```

${ShellId} expands to "Microsoft.PowerShell" in PowerShell.

Reply:
Ah good! So both parts of the input `${ShellId}` and `%COMSPEC%` are to detect windows shells.

Reply:
Correct.